### DISCUSSION TITLE
Can i change the bandwith dynamically

### DISCUSSION THREAD

**Discussion by mertadgn** [aiortc/aiortc#963] [Q&A]
Hi everyone!
In an example I made with aiortc, I take a video and send it to localhost:9006 with a python script and watch the video I sent. An html (js) code on the receiver side, as you can imagine.

What I want to do is to be able to actively change the resolution with a combobox on the page at localhost:9006.

For example, 720 360 240 options and less data usage in this way.

I found an example for this in webrtc (https://webrtc.github.io/samples/src/content/peerconnection/bandwidth/) but I couldn't find an example for it in aiortc. 

In a github discussion in 2020 that it is not possible to dynamically set the bitrate. (I searched again but couldn't find it, sorry)
Is it possible to do what I'm talking about now?

**Comment by Pharbi** [comment]
To achieve that goal, you can try to make updates to the resolution by renegotiating the offer that is sent over. In the webcam example there's this line in [client](https://github.com/aiortc/aiortc/blob/main/examples/webcam/client.js#L24) which is related to this line in [webcam.py](https://github.com/aiortc/aiortc/blob/main/examples/webcam/webcam.py#L63). Where it processes the request sent over and updates the video if given a `video_codec` field in the request. Shown [here](https://github.com/aiortc/aiortc/blob/9f14474c0953b90139c8697a216e4c2cd8ee5504/examples/webcam/webcam.py#L92C13-L92C24)
